code/SingleRecordAdmin.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Forms\Form;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * The base class for admin sections that exist for editing a single record.
+ */
+abstract class SingleRecordAdmin extends LeftAndMain
+{
+    /**
+     * Determines if there should be a single record in the database that this admin edits.
+     *
+     * If this is not true, you need to provide a mechanism to tell the form which record it should be editing.
+     * This could be an action (e.g. edit/$ID), or could be based on something about the current member, etc.
+     */
+    private static bool $only_one_record = true;
+
+    /**
+     * If no record exists, allow a new one to be created
+     */
+    private static bool $allow_new_record = true;
+
+    public function getEditForm($id = null, $fields = null): ?Form
+    {
+        $form = parent::getEditForm($id, $fields);
+        // Keep the tabs on the top row, rather than underneath
+        if ($form?->Fields()->hasTabSet()) {
+            $form->Fields()->findOrMakeTab('Root')->setTemplate('SilverStripe/Forms/CMSTabSet');
+        }
+        return $form;
+    }
+
+    public function getRecord($id): ?DataObject
+    {
+        if (!$id) {
+            return $this->getSingleRecord();
+        }
+        return parent::getRecord($id);
+    }
+
+    protected function getSingleRecord(): ?DataObject
```

Comment:
```suggestion
    private function getSingleRecord(): ?DataObject
```
If we're providing `only_one_record` and `allow_new_record` and implementing logic to do stuff based on how they're set, why would anyone need to override this in a subclass? Things should be private unless there's an immediate need to make it protected.

Reply:
Please see the docs where overriding this method is explicitly mentioned as a way to provide more complex behaviour for getting this record.